### ISSUE TITLE
"not a supported wheel on this platform" when installing local whl file

### ISSUE THREAD

**Issue by ed-alertedh** [pypa/pipenv#1159]
I'm trying to set up a basic numpy+scipy env on Windows using the builds from https://www.lfd.uci.edu/~gohlke/pythonlibs. I downloaded the latest wheels for numpy and scipy, then placed them in my project folder. Due to #1105 I currently have to use an absolute path and a "file://" URI to install from a local wheel. My Pipfile looks like this:

    [[source]]

    url = "https://pypi.python.org/simple"
    verify_ssl = true
    name = "pypi"

    [packages]

    "4dd5b0a" = {file = "file://d:/dev/branch_tests/issue-118/numpy-1.13.3+mkl-cp36-cp36m-win_amd64.whl", os_name = "=='nt'"}
    "3a45660" = {file = "file://d:/dev/branch-tests/issue-118/scipy-1.0.0-cp36-cp36m-win-amd64.whl", os_name = "=='nt'"}
    numpy = { version = "*", os_name = "=='linux'", index="pypi" }
    scipy = { version = "*", os_name = "=='linux'", index="pypi" }
    dask = "*"
    distributed = "*"
    dm-sonnet = "*"
    fastparquet = "*"
    pandas = "*"
    pyodbc = "*"
    scikit-learn = "*"
    seaborn = "*"
    sqlalchemy = "*"
    tensorflow-gpu = ">= 1.4.0"

    [dev-packages]

    pytest = "*"
    jupyter = "*"
    nbdime = "*"
    #pyreadline { version = "*", os_name = "=='nt'", index="pypi" }

    [requires]

    python_version = "3.6.3"

##### Describe your environment

1. Windows 10
1. Python version: 3.6.3 64-bit
1. Pipenv version: 8.3.2

##### Expected result

Numpy and scipy install successfully, plus all the other packages.

##### Actual result

Everything installs fine except for scipy, which fails with the message "not a supported wheel on this platform". If I run `pipenv run pip install scipy-1.0.0-cp36-cp36m-win-amd64.whl` then the package installs fine.

When I run with --verbose I get a couple of warnings which don't seem related:
"Config variable 'Py_DEBUG' is unset, Python ABI tag may be incorrect"
"Config variable 'WITH_PYMALLOC' is unset, Python ABI tag may be incorrect"

The contents of the temporary pipenv-*-requirements.txt files are:

    file://d:/dev/branch-tests/issue-118/scipy-1.0.0-cp36-cp36m-win-amd64.whl; os_name =='nt'

and

    file://d:/dev/branch_tests/issue-118/numpy-1.13.3+mkl-cp36-cp36m-win_amd64.whl; os_name =='nt'

##### Steps to replicate
1. (probably irrelevant) set PIPENV_VENV_IN_PROJECT=1
2. Download numpy-1.13.3+mkl-cp36-cp36m-win_amd64.whl and scipy-1.0.0-cp36-cp36m-win-amd64.whl
3. Update Pipfile to point to location of the downloaded wheels 
4. `pipenv install --verbose`

**Comment by ed-alertedh** [comment]
OK, just tracked down the actual issue. pipenv seems to be generating a slightly incorrect entry to add to my Pipfile. I added the entry using `pipenv.exe install file://D:/dev/branch_tests/issue-118/scipy-1.0.0-cp36-cp36m-win_amd64.whl`. It then generated the entry as above which has the suffix "win-amd64" instead of the correct suffix "win_amd64". I can manually edit Pipfile as a workaround for now, but this does seem to be a minor bug.

**Comment by kennethreitz** [comment]
Solution: install pipenv with python 3.

**Comment by ed-alertedh** [comment]
I did:

     >whereis pipenv
    C:\Python36-x64\Scripts\pipenv.exe

TL;DR pipenv will install a .whl from file:// but the Pipfile entry gets mangled. Details above.

**Comment by ed-alertedh** [comment]
update: now that #1105 is resolved a workaround for this is to avoid using file:// URI's. It's also possible the complete overhaul of this part of pipenv resolved this anyway

**Comment by techalchemy** [comment]
@ed-alertedh I suspect this will be resolved when #1354 is merged, a lot of commits were wiped due to a few essential breakages in our history and broken testing infrastructure.  As a result we decided to re-merge all the changes from baseline to get this working. Specifically this should be addressed by 773892b and 035b78c together

**Comment by ed-alertedh** [comment]
I'm much happier using a relative file path anyway but thanks for the info! If I remember I'll try to reproduce this apparent bug in the next release and report back.
